packages/edit-site/src/components/global-styles/font-sizes.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * WordPress dependencies
+ */
+import { privateApis as blockEditorPrivateApis } from '@wordpress/block-editor';
+import { __, sprintf } from '@wordpress/i18n';
+import {
+	__experimentalItemGroup as ItemGroup,
+	__experimentalVStack as VStack,
+	__experimentalHStack as HStack,
+	FlexItem,
+	Button,
+	Tooltip,
+} from '@wordpress/components';
+import { plus } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+const { useGlobalSetting } = unlock( blockEditorPrivateApis );
+import Subtitle from './subtitle';
+import { NavigationButtonAsItem } from './navigation-button';
+
+function getAvailableIndex( sizes ) {
+	// Filter the slug field that ends with custom-{number}.
+	const indexes = sizes.map( ( size ) => {
+		const match = size.slug.match( /custom-(\d+)$/ );
+		return match ? parseInt( match[ 1 ], 10 ) : 0;
+	} );
+	return Math.max( ...indexes ) + 1;
+}
```

Comment:
We should be able to use the existing [`getNewIndexFromPresets()` function](https://github.com/WordPress/gutenberg/blob/19307c0b0b93b04a58fc25b7d22d3bcf40fd1a94/packages/edit-site/src/components/global-styles/utils.js#L24).

Reply:
done here: [`8465ece` (#62328)](https://github.com/WordPress/gutenberg/pull/62328/commits/8465ece2b6a0d330737821f189e41e31ddd86715)